### PR TITLE
Provider

### PR DESCRIPTION
**Pull request by ZdorovenkoKyrylo**

### CODE REVIEW

File: OutOfSchool/OutOfSchool.DataAccess/Enums/Category.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+namespace OutOfSchool.Services.Enums;
+
+public enum Category
+{
+    Complex,
+    Profile, 
+    specialized,
```
Comment (alrom) [inline OutOfSchool/OutOfSchool.DataAccess/Enums/Category.cs:7]: First letter must be upper

File: OutOfSchool/OutOfSchool.DataAccess/Enums/LegalStatus.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+namespace OutOfSchool.Services.Enums;
+
+public enum LegalStatus
+{
+    separateLegalEntity,
```
Comment (alrom) [inline OutOfSchool/OutOfSchool.DataAccess/Enums/LegalStatus.cs:5]: Same

File: OutOfSchool/OutOfSchool.DataAccess/Repository/ProviderRepository.cs [hunk 3]
```diff
@@ -30,6 +30,18 @@ public ProviderRepository(OutOfSchoolDbContext dbContext)
     /// <returns>Bool.</returns>
     public bool ExistsUserId(string id) => db.Providers.Any(x => x.UserId == id);
 
+    /*async Task<Provider> IProviderRepository.GetByEmail(string email)
```
Comment (alrom) [inline OutOfSchool/OutOfSchool.DataAccess/Repository/ProviderRepository.cs:33]: You could delete your comments )

File: OutOfSchool/OutOfSchool.WebApi/Services/ProviderTypeService.cs [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        this.localizer = localizer ?? throw new ArgumentNullException(nameof(localizer));
+        this.repository = repository ?? throw new ArgumentNullException(nameof(repository));
+        this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        this.mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+    }
+
+    /// <inheritdoc/>
+    public async Task<IEnumerable<ProviderTypeDto>> GetAll()
+    {
+        logger.LogInformation("Getting all Social Groups started.");
```
Comment (alrom) [inline OutOfSchool/OutOfSchool.WebApi/Services/ProviderTypeService.cs:36]: Social groups???? Check your comments )))

**Review by andriy-korchynskyy — APPROVED** [review]

**Pull request closed without merging**